crates/trident_api/src/config/host/health.rs

```
@@ -0,0 +1,210 @@
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+
+use crate::config::host::scripts::{Script, ServicingTypeSelection};
+use crate::status::ServicingType;
+
+/// Scripts that can be run on the host during Trident stages.
+/// These scripts are run in the order they are defined.
+/// Ensure that the scripts are idempotent as they may be run multiple times.
+#[derive(Serialize, Deserialize, Debug, Default, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+pub struct Health {
+    /// Scripts to be run before Trident commits a serviced target OS as 'provisioned'.  If any of
+    /// the scripts fail, commit will not be completed and, for A/B update, a rollback will be
+    /// triggered.
+    ///
+    /// These scripts run for installs and A/B updates. If runOn is specified for anything other
```

Should we error out instead of ignoring? Also iirc we have `allServicingTypes` selection or something like that

nit: `runOn`